src/jloptions.c

```diff
@@ -439,15 +439,20 @@ JL_DLLEXPORT void jl_parse_opts(int *argcp, char ***argvp)
             break;
         case opt_banner: // banner
             if (!strcmp(optarg, "yes"))
-                jl_options.banner = 1;
+                jl_options.banner = 10;
             else if (!strcmp(optarg, "no"))
                 jl_options.banner = 0;
             else if (!strcmp(optarg, "auto"))
                 jl_options.banner = -1;
-            else if (!strcmp(optarg, "short"))
-                jl_options.banner = 2;
-            else
-                jl_errorf("julia: invalid argument to --banner={yes|no|auto|short} (%s)", optarg);
+            else {
+                errno = 0;
+                int bval = strtol(optarg, NULL, 10);
+                if (errno == ERANGE)
+                    jl_errorf("julia: invalid argument to --banner={yes|no|0-100|auto} (%s)", optarg);
```

Comment:
The 0-100 option is not self explanatory, it should be documented.

If we're making banner size adaptive then maybe we should remove the short option without replacement.

Reply:
I thought I had managed to remove the short option here? Are there some spots I've missed?

Reply:
"Without replacement" meant don't add a command line option to specify banner size at all, simply always make it adaptive. But now that I know the size option is ordinal, I think that is a more reasonable banner size specification and probably worth including as you have already done.